### ISSUE TITLE
Naming a newly created folder renames all selected files instead

### ISSUE THREAD

**Issue by kettenfett** [tablacus/TablacusExplorer#594]
Pretty much what the titel said. 

It happens, when I use my "New Folder" shortcut, which I created like this:
![grafik](https://user-images.githubusercontent.com/23098362/167305679-0a921042-ffdd-44e5-a5a9-d88bf4617b9c.png)

Tablacus will create a new folder asking for a new name (everything like it should be).

- _(After pressing the New folder shortcut while files are selected)_
    - ![grafik](https://user-images.githubusercontent.com/23098362/167305913-96ea8da2-5121-4be9-bb1f-5334a50ca74b.png)

But then, upon entering a name and pressing Enter, all currently selected files are renamed instead.

- _(Entering a name)_
    - ![grafik](https://user-images.githubusercontent.com/23098362/167305941-b39cc112-263b-441e-9928-fde0f9127f4e.png)
- _(Result)_
    - ![grafik](https://user-images.githubusercontent.com/23098362/167305946-b9576485-f6f3-4211-83b7-c899d71e32a4.png)

The difference between my "New Folder" shortcut and using the context menu to create a new folder is precisely that, with the first method the active file selection stays intact, which I like.

**In short: Can you please make the Enter key just rename the new folder and keep the current file selection, AND NOT rename the files instead of the folder? Thank you.**

**Comment by tablacus** [comment]
Please try this script.
https://gist.github.com/tablacus/ee77b6c9018e8267e419cac6e54e9e1b

Regards,

**Comment by kettenfett** [comment]
> Please try this script. https://gist.github.com/tablacus/ee77b6c9018e8267e419cac6e54e9e1b
> 
> Regards,

It works to make a new folder. It shows a dialog box when I use the button, just like F7. Is there any difference to using the script and using F7?

Is the bug fixable while keeping the other method (without dialog box)?

**Comment by tablacus** [comment]
It's probably not a bug, it's in Microsoft's intended specification, so changing it would be difficult.

**Comment by kettenfett** [comment]
Whats the difference of using your script and F7?

**Comment by kettenfett** [comment]
I'm got an error "path not found" using the script while in the Windows "Downalod" folder:
![grafik](https://user-images.githubusercontent.com/23098362/168444478-b94c9f29-7811-4db0-ac2c-87aabc25ae25.png)

**Comment by tablacus** [comment]
I have fixed it.
Please try replacing it.
https://gist.github.com/tablacus/ee77b6c9018e8267e419cac6e54e9e1b

Regards,

**Comment by kettenfett** [comment]
Thank you, it's working!

**Comment by kettenfett** [comment]
Hi @tablacus ,
recently the error message i posted above has resurfaced (with 22.6.27 or prior). It only happens when trying to create a new folder in the windows "Downloads" folder. 

Sadly, I don't know how to debug the Script. Could you please provide me an updated version?

The Script I use for the Button is unchanged:
`const currentPath = FV.FolderItem.Path;
InputDialog("New Folder", "", function (r) {
  if (r) {
    let path = r.replace(/^\s+/, "");
    if (!/^[A-Z]:\\|^\\/i.test(path)) {
      path = BuildPath(currentPath, path);
    }
    fso.CreateFolder(path);
  }
});`
